File: annotation-processor/src/main/java/me/bristermitten/mittenlib/annotations/util/PrivateAnnotations.java
package me.bristermitten.mittenlib.annotations.util;

import java.util.Set;
import me.bristermitten.mittenlib.collections.Sets;
import me.bristermitten.mittenlib.config.names.ConfigName;

public class PrivateAnnotations {
    private static final Set<String> PRIVATE_ANNOTATIONS = Sets.of(ConfigName.class.getName());

    public static boolean isPrivate(String annotation) {
        return PRIVATE_ANNOTATIONS.contains(annotation);
    }
}

File: annotation-processor/src/main/scala/me/bristermitten/mittenlib/annotations/util/PrivateAnnotations.scala
package me.bristermitten.mittenlib.annotations.util

import me.bristermitten.mittenlib.config.names.ConfigName

object PrivateAnnotations:
  private val PRIVATE_ANNOTATIONS = Set(
    classOf[ConfigName].getName,
    classOf[Override].getName
  )

  def isPrivate(annotation: String): Boolean =
    PRIVATE_ANNOTATIONS.contains(annotation)
